backend/src/akvo/lumen/lib/aggregation/pie.clj

```diff
@@ -17,10 +17,24 @@
         bucket-column-name (get bucket-column "columnName")
         bucket-column-title (get bucket-column "title")
         bucket-column-type (get bucket-column "type")
-        counts (run-query tenant-conn table-name bucket-column-name filter-sql)]
-    (lib/ok {"metadata" {"bucketColumnTitle" bucket-column-title
-                         "bucketColumnType" bucket-column-type}
-             "data" (mapv (fn [[bucket-value bucket-count]]
-                            {"bucketValue" bucket-value
-                             "bucketCount" bucket-count})
-                          counts)})))
+        counts (run-query tenant-conn table-name bucket-column-name filter-sql)
+        max-segments 50]
+    (if (> (count counts) max-segments)
+      (lib/ok
```

Comment:
Does this mean the client expect a 200 response?

Reply:
Yeah, I wasn't sure about this. Currently the client expects a 200. Perhaps another type of response would be more semantically correct. It's not really an error or a bad request. I gues a `413 - Payload Too Large` might fit, but my impression was that's more for situations where the request itself is too large. A 4xx code feels weird because the client can't know ahead of time anything about the aggregation. Maybe a generic `500`? 

A `418` also seems a possibility

Reply:
418 sounds like a good option if we want to be semantically correct, otherwise, I'm fine with whatever. I mean it's a bit odd to have an error with a 200 :-). I would have no problem responding with a 400, but it's as I said not the end of the world. Noticed that some of the other aggregations don't have multiple response options at all. Since the client does a fair bit of checking I'm not sure we need to cover all cases. But maybe we can be a bit pragmatic and expand where needed, moving forward.

Reply:
Yes, for now we only respond with an error when the aggregation response will result in too many elements for a given chart type, and for now this only applies to pie/donut charts and bar charts.

I changed the response to a generic "bad-request" as already implemented and used in other places. After some thought this seems better than a 200, which I agree is weird